mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -89,22 +94,30 @@ public void preCreateTable(org.apache.hadoop.hive.metastore.api.Table hmsTable)
     }
 
     // If the table does not exist collect data for table creation
-    String schemaString = catalogProperties.getProperty(InputFormatConfig.TABLE_SCHEMA);
-    Preconditions.checkNotNull(schemaString, "Please provide a table schema");
-    // Just check if it is parsable, and later use for partition specification parsing
-    Schema schema = SchemaParser.fromJson(schemaString);
-
-    String specString = catalogProperties.getProperty(InputFormatConfig.PARTITION_SPEC);
-    if (specString != null) {
-      // Just check if it is parsable
-      PartitionSpecParser.fromJson(schema, specString);
-    }
+    // - InputFormatConfig.TABLE_SCHEMA, InputFormatConfig.PARTITION_SPEC takes precedence so the user can override the
+    // Iceberg schema and specification generated by the code
+    // - Partitioned Hive tables are currently not allowed
+
+    Schema schema = schema(catalogProperties, hmsTable);
+    PartitionSpec spec = spec(schema, catalogProperties);
+
+    catalogProperties.put(InputFormatConfig.TABLE_SCHEMA, SchemaParser.toJson(schema));
+    catalogProperties.put(InputFormatConfig.PARTITION_SPEC, PartitionSpecParser.toJson(spec));
+
+    Preconditions.checkArgument(hmsTable.getPartitionKeys() == null || hmsTable.getPartitionKeys().isEmpty(),
+        "Partitioned Hive tables are currently not supported");
```

Comment:
I think this check should be in `spec`, like validation that the table schema is not present.

Reply:
@rdblue: I am afraid I don't get what you mean here by "this check should be in `spec`". You simply mean that we should document it, or you have something else in your mind?

Could you please check the other comments too? This one is blocking writes too long...

Thanks, Peter

Reply:
I think what @rdblue means here is that it should be in the `spec()` method below

Reply:
Ohh.. I get it. Thanks!
Done